Tipsy/Tipsy/Controllers/CalculatorViewController.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  ViewController.swift
+//  Tipsy
+//
+//  Created by Angela Yu on 09/09/2019.
+//  Copyright © 2019 The App Brewery. All rights reserved.
+//
+
+import UIKit
+
+class CalculatorViewController: UIViewController {
+    
+    var totalCost = 0.0
+    var numberofPeople = 2
+    var tipRatio = 0.0
+    var result = "0.0"
+    
+    @IBOutlet weak var billTextField: UITextField!
+    @IBOutlet weak var zeroPctButton: UIButton!
+    @IBOutlet weak var tenPctButton: UIButton!
+    @IBOutlet weak var twentyPctButton: UIButton!
+    @IBOutlet weak var splitNumberLabel: UILabel!
+    
+    @IBAction func tipChanged(_ sender: UIButton) {
+        billTextField.endEditing(true)
+        if sender.currentTitle == "0%" {
+            tipRatio = 0.0
+            zeroPctButton.isSelected = true
+            tenPctButton.isSelected = false
+            twentyPctButton.isSelected = false
+        } else if sender.currentTitle == "10%" {
+            tipRatio = 0.1
+            zeroPctButton.isSelected = false
+            tenPctButton.isSelected = true
+            twentyPctButton.isSelected = false
+        } else if sender.currentTitle == "20%" {
+            tipRatio = 0.2
+            zeroPctButton.isSelected = false
+            tenPctButton.isSelected = false
+            twentyPctButton.isSelected = true
+        }
+        
+    }
+    
+    
+    @IBAction func stepperValueChanged(_ sender: UIStepper) {
+        splitNumberLabel.text = String(format: "%.0f", sender.value)
+    }
+    
+    @IBAction func calculatePressed(_ sender: UIButton) {
+        if billTextField.text != "" {
```

Comment:
Swiftでは以下のように書くのが一般的です。

* メソッドの事前条件はguardでチェックして早期リターンする
* 空文字判定はisEmptyを使用

```suggestion
        guard !billTextField.text.isEmpty else { return }
```

Reply:
Nil coalescing operatorも使うと、以下のように記述量を減らすことができそうです。

```swift
    @IBAction func calculatePressed(_ sender: UIButton) {
        guard let totalCost = Double(billTextField.text ?? ""),
              let numberofPeople = Double(splitNumberLabel.text ?? "") else { return }

        result = String(format: "%.2f", (totalCost + totalCost * tipRatio) / numberofPeople)
        self.performSegue(withIdentifier: "goToResult", sender: self)
    }
```

強制アンラップを使わずに、アンラップする方法は以下を参考にしてください。
https://techblog.recochoku.jp/8058